src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -652,6 +660,13 @@ export default class FilingHistoryList extends Vue {
             dissolutionType: filing?.data?.dissolution?.dissolutionType })
       }
 
+      // add properties for limited restorations
+      if (this.isTypeLimitedRestoration(filing)) {
+        item.isTypeLimitedRestoration = true
+        item.legalName = filing?.data?.restoration?.legalName
+        item.expiryDate = filing?.data?.restoration?.expiryDate
+      }
```

Comment:
The Typescript [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) operator is a double-edged sword -- it can mask errors that you should be handling (in previous code).

Here, you don't have to check that filing is non-null/undefined because that was checked previously, so it's guaranteed to be OK.

I looked at other code in here, and it doesn't use ? on data either, though maybe it should.

Maybe a better approach here, for this type of filing, is to verify that filing.data.restoration is non-null/undefined (as that would be a real error) and throw an exception otherwise (which would get caught by line 671).

Reply:
Good call.  I've changed it as you suggest.